server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/ElementaryNodes.java

```diff
@@ -0,0 +1,130 @@
+  private static final String SIGN_CLAUSE = "SIGN CLAUSE";
+  private final boolean isBlankWhenZeroPresent;
+  private final boolean isSignClausePresent;
+  EffectiveDataType effectiveDataType;
+  String picClause;
+  @Getter UsageFormat usageFormat;
+
+  protected ElementaryNodes(
+      Locality location,
+      int level,
+      String name,
+      VariableType variableType,
+      boolean global,
+      boolean isBlankWhenZeroPresent,
+      boolean isSignClausePresent,
+      String picClause,
+      UsageFormat usageFormat) {
+    super(location, level, name, variableType, global);
+    this.isBlankWhenZeroPresent = isBlankWhenZeroPresent;
+    this.isSignClausePresent = isSignClausePresent;
+    this.picClause = picClause;
+    this.usageFormat = usageFormat;
+    this.effectiveDataType = setEffectiveDataType();
+  }
+
+  private EffectiveDataType setEffectiveDataType() {
+    return Arrays.stream(EffectiveDataType.values())
+        .filter(type -> type.checkDataType(this))
+        .findFirst()
+        .orElse(UNDETERMINED);
+  }
+
+  /**
+   * Validate proper use of PIC and USAGE clause for elementary nodes
+   *
+   * @return SyntaxError
+   */
+  protected List<SyntaxError> validate() {
+    List<SyntaxError> errors = new ArrayList<>();
+    validatePicClauseUsage().unwrap(errors::addAll);
+    validateCompatibleUsageClause().unwrap(errors::addAll);
+    return errors;
+  }
+
+  private ResultWithErrors<Boolean> validateCompatibleUsageClause() {
+    boolean check = isSignClausePresent || isBlankWhenZeroPresent;
+    if (Objects.isNull(usageFormat)) return FALSE_RESULT_WITH_NO_ERRORS;
+    if (VariableDefinitionUtil.BLANK_AND_SIGN_NOT_ALLOWED_USAGE.contains(usageFormat) && check)
+      return new ResultWithErrors<>(
+          true,
+          ImmutableList.of(
+              this.getError(
+                  MessageTemplate.of(
+                      SEMANTIC_IMPROPER_USE_BLANK_WHEN_ZERO_AND_SIGN_CLAUSE, getInCompatibleClause()),
+                  ErrorSeverity.WARNING)));
+    return FALSE_RESULT_WITH_NO_ERRORS;
+  }
+
+  private String getInCompatibleClause() {
+    if (isBlankWhenZeroPresent) return BLANK_WHEN_ZERO;
+    return SIGN_CLAUSE;
+  }
+
+  private ResultWithErrors<Boolean> validatePicClauseUsage() {
+    if (StringUtils.isBlank(getPicClause())) return FALSE_RESULT_WITH_NO_ERRORS;
+    if (VariableDefinitionUtil.NO_PIC_CLAUSE_USAGE.contains(getUsageFormat()))
+      return new ResultWithErrors<>(
+          true,
+          ImmutableList.of(
+              this.getError(
+                  MessageTemplate.of(SEMANTICS_NO_PIC_CLAUSE, getUsageFormat().toString()),
+                  ErrorSeverity.WARNING)));
+
+    Function<ElementaryNodes, ResultWithErrors<Boolean>> orDefault =
+        VariableDefinitionUtil.PIC_AND_USAGE_CLAUSE_VALIDATOR.getOrDefault(
+            effectiveDataType, node -> new ResultWithErrors<>(false, ImmutableList.of()));
+
+    return orDefault.apply(this);
```

Comment:
Looks like an overcomplication for this purpose

Reply:
Done.